src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackDecoder.cs

```diff
@@ -397,14 +545,14 @@ public void CompleteDecode()
 
         private void OnIndexedHeaderField(int index, IHttpHeadersHandler? handler)
         {
-            HeaderField header = GetHeader(index);
+            ref readonly HeaderField header = ref GetHeader(index);
             handler?.OnHeader(header.Name, header.Value);
             _state = State.Ready;
```

Comment:
The GetHeader call is only necessary if we're going to call OnHeader, so seems like this would be better as:
```C#
if (handler != null)
{
    ref readonly HeaderField header = ref GetHeader(index);
    handler.OnHeader(header.Name, header.Value);
}
_state = State.Ready;
```